src/rdkafka_request.c

```diff
@@ -3527,6 +3527,116 @@ rd_kafka_DeleteTopicsRequest (rd_kafka_broker_t *rkb,
 }
 
 
+/**
+ * @brief Construct and send DeleteRecordsRequest to \p rkb
+ *        with the DeleteRecords objects (DeleteRecords_t *) in
+ *        \p delete_records, using \p options.
+ *
+ *        The response (unparsed) will be enqueued on \p replyq
+ *        for handling by \p resp_cb (with \p opaque passed).
+ *
+ * @returns RD_KAFKA_RESP_ERR_NO_ERROR if the request was enqueued for
+ *          transmission, otherwise an error code and errstr will be
+ *          updated with a human readable error string.
+ */
+rd_kafka_resp_err_t
+rd_kafka_DeleteRecordsRequest (rd_kafka_broker_t *rkb,
+                              const rd_list_t *offsets_list /*(rd_kafka_topic_partition_list_t*)*/,
+                              rd_kafka_AdminOptions_t *options,
+                              char *errstr, size_t errstr_size,
+                              rd_kafka_replyq_t replyq,
+                              rd_kafka_resp_cb_t *resp_cb,
+                              void *opaque) {
+        rd_kafka_buf_t *rkbuf;
+        int16_t ApiVersion = 0;
+        size_t of_TopicArrayCnt = 0, of_PartArrayCnt = 0;
+        int32_t topic_cnt = 0, part_cnt = 0;
+        const char *last_topic = "";
+        int features;
+        int i = 0;
+        rd_kafka_topic_partition_list_t *partitions;
+        int op_timeout;
+
+        partitions = rd_list_elem(offsets_list, 0);
+
+        if (partitions == NULL) {
+                rd_snprintf(errstr, errstr_size, "Partition vector is NULL");
```

Comment:
This should be checked in the Admin API, by the time we get here input params should be proper.